### PR TITLE
Added functionality to fix issue #9

### PR DESCRIPTION
**Pull request by TimVerhaegen**
Added a simple item called gloves that lets you receive berries when breaking the bush while holding the item.

To test:

1. Run the game
2. Go into the console and give yourself some seeds. (`give AppleSeed` for instance)
3. Plant the seeds (at least 2)
4. Wait for them to grow and have produce
5. Break one not using the gloves
6. Break one using the gloves (`give Gloves`)

You'll receive 1-3 Apples AND seeds in case you used the gloves and only seeds when you haven't.

### CODE REVIEW

**Comment by jellysnake** [comment]
So I actually considered having the bush drop the produce on being broken and decided to add it as an option.

 So if the `sustainable` field of the `BushDefinitionComponent` is set to be `false` then the bush will break on being harvested and drop the seeds as well. Try harvesting the potato to see this in action.

File: src/main/java/org/terasology/simpleFarming/systems/BushAuthoritySystem.java [hunk 1]
```diff
@@ -115,6 +119,11 @@ public void onPlantDestroyed(DoDestroyPlant event, EntityRef entity, BushDefinit
         } else {
             onBudDestroyed(bushComponent, event.isParentDead);
         }
+        EntityRef selected = localPlayer.getCharacterEntity().getComponent(CharacterHeldItemComponent.class).selectedItem;
+        boolean b = selected.getParentPrefab().getName().equalsIgnoreCase("SimpleFarming:gloves");
```
Comment (aaron-harris) [inline src/main/java/org/terasology/simpleFarming/systems/BushAuthoritySystem.java]: I think this condition should either be inside the if statement (`if(selected.getParentPrefab().`...`) {`) (my preference), or it should have a more descriptive name like `glovesUsed`.

File: src/main/java/org/terasology/simpleFarming/systems/BushAuthoritySystem.java [hunk 2]
```diff
@@ -115,6 +119,11 @@ public void onPlantDestroyed(DoDestroyPlant event, EntityRef entity, BushDefinit
...
+        if(b){
```
Comment (aaron-harris) [inline src/main/java/org/terasology/simpleFarming/systems/BushAuthoritySystem.java]: I think this condition should either be inside the if statement (`if(selected.getParentPrefab().`...`)`) (my preference), or it should have a more descriptive name like `glovesUsed`.

**Comment by aaron-harris** [comment]
I think this points to the need for a more general system allowing the tool used to harvest and/or break a plant to intercede in the harvest process.  I don't know how that system would look, or if this is the right module for it to live in, but starting off by special-casing individual items seems like getting off on the wrong foot.

That said, if this is useful now, it can sit around until we have the system built, or even serve as a prototype/test case as we build the system around it, maybe along with some other tools that do different things.

**Comment by jellysnake** [comment]
I fundamentally believe that this specific functionality is not needed. If a plant is to drop produce when broken then doesn't the `sustainable` field do just that, as it means the plant is removed and seeds are dropped when it is harvested. (Eg. potato plant).  
On a similar vein why do you need the plant to drop produce when broken? From a balance perspective it's very easy to grow a large number of bushes quickly already, I don't see the requirement for this item.

If this functionality is desired I think it should be added as an option on the prefab, perhaps `produceOnBreak` or a better name. Or alternatively added to the `sustainable` option as it already implements a very similar functionality. An item could still give this functionality by using higher priority on than the method in SimpleFarming and simple enable the flag in the prefab.

File: src/main/java/org/terasology/simpleFarming/systems/BushAuthoritySystem.java (hunk 2 above)
Comment (TimVerhaegen) [reply src/main/java/org/terasology/simpleFarming/systems/BushAuthoritySystem.java]: Well, personally I prefer keeping it short but that'd be a good name indeed.

Comment (jellysnake) [reply src/main/java/org/terasology/simpleFarming/systems/BushAuthoritySystem.java]: Whilst keeping it short is good, there is no difference between using `b` and `theBooleanThatIndicatesIfTheGlovesWereUsed` to the compiler. Which means that it's better to use something that makes it very clear what the boolean does.

Comment (Cervator) [reply src/main/java/org/terasology/simpleFarming/systems/BushAuthoritySystem.java]: By the way try to run Checkstyle on affected files like this, it'll probably both highlight the naming issue as well as some whitespace stuff that helps the code style stay consistent :-)

(There is a Checkstyle plugin for IntelliJ if you haven't set it up yet)

**Comment by Cervator** [comment]
One bit I'm confused about - you generally harvest bushes and things for produce via `e` click, while breaking a bush gives seeds. Additional nice options (like somewhat talked about in #9) would be to have tools that enhance the process, such as increasing the yield, or getting seeds without breaking a bush.

Doing both in one action seems counter-intuitive to me - a player would probably be out for either produce *or* seeds, but not necessarily both at once?

Whether ripe fruit would drop on a bush block being broken does seem kinda like a whole other system that would be applied in general. Where it is more like an incidental effect than purposeful :-)

**Pull request closed without merging**